src/sedtrails/transport_converter/format_converter.py

```diff
@@ -1,80 +1,544 @@
+        
+        # Create a dictionary with time-specific data
+        time_data = {
+            'time': self.times[time_idx],
+            'reference_date': self.reference_date,
+            'x': self.x,
+            'y': self.y,
+            'bed_level': self.bed_level,  # Bed level is typically time-independent
+            'fractions': self.fractions,
+            
+            # Extract time slice from time-dependent variables
+            'water_depth': self.water_depth[time_idx],
+            'mean_bed_shear_stress': self.mean_bed_shear_stress[time_idx],
+            'max_bed_shear_stress': self.max_bed_shear_stress[time_idx],
+            'sediment_concentration': self.sediment_concentration[time_idx],
+            
+            # Extract time slice from vector quantities
+            'depth_avg_flow_velocity': {
+                'x': self.depth_avg_flow_velocity['x'][time_idx],
+                'y': self.depth_avg_flow_velocity['y'][time_idx],
+                'magnitude': self.depth_avg_flow_velocity['magnitude'][time_idx]
+            },
+            'bed_load_sediment': {
+                'x': self.bed_load_sediment['x'][time_idx],
+                'y': self.bed_load_sediment['y'][time_idx],
+                'magnitude': self.bed_load_sediment['magnitude'][time_idx]
+            },
+            'suspended_sediment': {
+                'x': self.suspended_sediment['x'][time_idx],
+                'y': self.suspended_sediment['y'][time_idx],
+                'magnitude': self.suspended_sediment['magnitude'][time_idx]
+            },
+            'nonlinear_wave_velocity': {
+                'x': self.nonlinear_wave_velocity['x'][time_idx],
+                'y': self.nonlinear_wave_velocity['y'][time_idx],
+                'magnitude': self.nonlinear_wave_velocity['magnitude'][time_idx]
+            }
+        }
+        
+        return time_data
+    
+    def __len__(self) -> int:
+        """
+        Get the number of time steps.
+        
+        Returns:
+        --------
+        int
+            Number of time steps
+        """
+        return len(self.times)
+    
+    def get_time_index(self, target_time: float) -> int:
+        """
+        Find the index of the closest time to the target time.
+        
+        Parameters:
+        -----------
+        target_time : float
+            Target time in seconds since reference_date
+            
+        Returns:
+        --------
+        int
+            Index of the closest time
+        """
+        # Find the closest time
+        time_diffs = np.abs(self.times - target_time)
+        return np.argmin(time_diffs)
+    
+    def get_interpolation_indices(self, target_time: float) -> Tuple[int, int, float]:
+        """
+        Get indices for interpolation between two time steps.
+        
+        Parameters:
+        -----------
+        target_time : float
+            Target time in seconds since reference_date
+            
```

Comment:
I have de feeling that this function should belong somewhere else, because it is part of the interpolation.